lib/msf/core/rpc/v10/rpc_db.rb

```diff
@@ -276,15 +276,31 @@ def rpc_creds(xopts)
           port = login.service.port.to_i
           proto = login.service.proto.to_s
         end
-        ret[:creds] << {
+
+        if cred.private.nil?
+            ret[:creds] << {
+                :user => cred.public.username.to_s,
+                :pass => nil,
+                :updated_at => cred.public.updated_at.to_i,
+                :type => nil,
+                :host => host,
+                :port => port,
+                :proto => proto,
+                :sname => sname
+            }
+        else
+            ret[:creds] << {
                 :user => cred.public.username.to_s,
                 :pass => cred.private.data.to_s,
                 :updated_at => cred.private.updated_at.to_i,
                 :type => cred.private.type.to_s,
                 :host => host,
                 :port => port,
                 :proto => proto,
-                :sname => sname}
+                :sname => sname
+            }
+        end
+
```

Comment:
consider instead of building two separate maps and having to maintain each

```
current_pass = nil
current_type = nil
unless cred.private.nil?
  current_pass = cred.private.data.to_s
  current_type = cred.private.type.to_s
end

ret[:creds] << {
  :user => cred.public.username.to_s,
  :pass => current_pass,
  :updated_at => cred.public.updated_at.to_i,
  :type => current_type,
  :host => host,
  :port => port,
  :proto => proto,
  :sname => sname
}
```

using ternary operations for the two specific keys would be a common path also
```
:pass => cred.private.nil? ? nil : cred.private.data.to_s
```